### DISCUSSION TITLE
tailwind css installed but not working in my vite + react project

### DISCUSSION THREAD

**Discussion by rmGFW** [tailwindlabs/tailwindcss#12312] [Help] [answered]
**What version of Tailwind CSS are you using?**
3.3.5

**What build tool (or framework if it abstracts the build tool) are you using?**

vite

**What version of Node.js are you using?**

v20.6.1

**What browser are you using?**

chrome

**What operating system are you using?**

macos

**Describe your issue**

my project use vite + react (i create project select vanilla js and after add react to it )
i install the tailwind css follow the guide https://tailwindcss.com/docs/guides/vite 
no any step is missing , but the tailwindcss seems not work, 

look this screenshot, none class's css can be find in the style pane
<img width="1330" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/66858140/63963813-2a65-419e-8e6c-7740e5552f7c">

**Comment by thecrypticace** [comment, marked as answer]
Hey! Did you update your content config? It looks to me like utilities aren't being generated because it isn't scanning any files.

**Reply by rmGFW** [reply]
sure , here is the content 
```js
/** @type {import('tailwindcss').Config} */
export default {
  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
  theme: {
    extend: {},
  },
  plugins: [],
};

```

**Reply by thecrypticace** [reply]
Can you package up the project in a git repo and put it on github to look at?

**Reply by MaxwellPechman** [reply]
Have you guys found the answer? I am currently having a similiar problem. I set up a project using vite + react-ts and tailwindcss and did everything according to the docs but it is not working for me. It has worked with older versions of vite though. Could you help me? I was also using pnpm when using vite and installing tailwindcss.

**Reply by BLOW3ED** [reply]
hello there, i have exactly the same problem , have you found the way to make it run succesfully?

**Reply by amoeba25** [reply]
I had the exact same problem. I had changed my vite.config.ts file to this and it worked for me

```export default defineConfig({
  plugins: [react()],
  css: {
    postcss: {
      plugins: [tailwindcss()],
    },
  }
})```

**Reply by theactualdev** [reply]
Did you have to install import tailwindcss to the vite.config.js file?

I have this vite repo and I'm facing the same issue on this repo:

https://github.com/theactualdev/react-calculator

**Reply by amoeba25** [reply]
Yes, here is the full vite.config.js that I used 

```
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import tailwindcss from 'tailwindcss'


export default defineConfig({
  plugins: [react()],
  css: {
    postcss: {
      plugins: [tailwindcss()],
    },
  }
})
```

**Reply by theactualdev** [reply]
It still doesn't work for some reason, can you help me check out my repo?

**Reply by amoeba25** [reply]
I had a look, changing the tailwind.config.js to this did the trick. I am not sure why it requires the ts and tsx files (might be something to do with if you installed the JS or the TS version of files from vite?) but try it out

```
/** @type {import('tailwindcss').Config} */
export default {
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

**Reply by theactualdev** [reply]
It works, thanks a bunch bro!

**Reply by Achmad-M** [reply]
thanks

**Reply by ribhuSingh** [reply]
Thank you very much

**Reply by iamgabrielsoft** [reply]
@amoeba25 solution works really well

**Reply by MaheshPawaar** [reply]
I had same issue. but tweaking vite.config.js helped me. Thanks!

**Reply by jimmyzzxhlh** [reply]
Ran into the same issue. This really needs to be documented in the official documentation: https://tailwindcss.com/docs/guides/vite

**Reply by MuhammadOwais-SE** [reply]
@jimmyzzxhlh jimmyzzxhlh Thank you. Your answer help me.

**Reply by dzienisz** [reply]
True, it's needs to be added to the documentation.

**Reply by ChauManh** [reply]
following document is true https://tailwindcss.com/docs/installation/using-vite.
config it doesn't matter.
you should delete App.css and clear default css in index.css, just patse @import 'tailwindcss'; in this file

**Reply by myselfsiddharth** [reply]
@amoeba25 That worked. Thankyou so much !

**Reply by Dhruvp33** [reply]
> Ran into the same issue. This really needs to be documented in the official documentation: https://tailwindcss.com/docs/guides/vite

brooooooo @jimmyzzxhlh omg man thankyou sooo much broo i was stuck on the same issue for more than 24 hours thankyou so much buddy

**Comment by manish0kuniyal** [comment]


**Reply by YuZhang-steven** [reply]
Works for me too. 
No idea what happend

**Reply by heyvipul** [reply]
it works thank you so much!

**Reply by Mightyshambel** [reply]
it worked thanks

**Reply by JalalJonaid** [reply]
Wonderful. Thanks for the tip.

**Reply by shugi12345** [reply]
finally! thank you very much!!

**Reply by Sarbpreetsingh99** [reply]
worked for me,
thanks mate.

**Reply by ichensun** [reply]
Excellent! Works for me thanks

**Reply by Janani-Balasooriya** [reply]
Many thanks! This works!!

**Reply by Writwiks** [reply]
Thank you so much buddy after an hour finally solved the issue.

**Reply by MVKarthikReddy** [reply]
That is (using cdn)  not working after deploying. 

I followed the tailwind css guide but it is also not working.
Can anyone help me?

**Reply by amoeba25** [reply]
Tailwind doesn't recommend using the play CDN for production. Here from their docs
"Use the Play CDN to try Tailwind right in the browser without any build step. The Play CDN is designed for development purposes only, and is not the best choice for production.". You will have to tweak the vite config to make it work

**Reply by SubhanKashif7** [reply]
> Hi
> 
> you may also try updating `index.html`
> 
> try adding this script tag
> 
> ```
> <script src="https://cdn.tailwindcss.com"></script>
> ```
> 
> it worked for me

this is not for production

**Reply by boiwantlearncode** [reply]
Am I the only one not seeing anything?

**Reply by sagarsingh0001** [reply]
This worked for me . thanks!!

**Comment by iamtanmoy2002** [comment]
hey, my tailwind is working but suggestion is not working, i have the tailwind intelligence installed but still I'm not getting any  suggestions. Need help.

**Reply by PRRASHANT** [reply]
press ctrl+space when you will write the css part

**Reply by raghavvvgaba** [reply]
Why isn't it showing automatically? It's irritating that I have to press ctrl+space for each class that I write

**Comment by TinySkillet** [comment]
Make sure you don't have spaces here:

`"./src/**/*.{js,ts,jsx,tsx}",`

I apparently had it as `"./src/**/*.{js ,ts, jsx, tsx}",`  lol

**Reply by Sophidoo** [reply]
This really helped

**Reply by 3astW1nd** [reply]
Thank you

**Comment by Kvngstar** [comment]
@amoeba25 thanks so much for the react-ts solution. it worked

**Comment by na-zi-ya** [comment]
I'm having trouble getting Tailwind CSS to work in my Vite + React project. I have set up my vite.config.js and tailwind.config.js files as follows:
![image](https://github.com/user-attachments/assets/98e1c430-a51a-4194-9c6c-40987b9a2b44)
![image](https://github.com/user-attachments/assets/bb287f10-da24-4abf-a849-56962087f3a6)
I have included Tailwind CSS classes in my components, but they don't seem to be applying,imported the Tailwind CSS styles in my main CSS file (index.css)
Could someone please help me identify what might be going wrong or suggest any steps I might have missed?

**Reply by elishamugenyi** [reply]
If you have this tailwind.config.js, means you have tailwindcss installed. Step 2, install @tailwindcss/vite (npm i @tailwindcss/vite). Go to your vite.config.ts file ad add this line (import tailwindcss from '@tailwindcss/vite' ) but without those brackets. Still inside vite.config.ts, in the export default defineConfig{}, add tailwindcss(), like this, export default defineConfig({
  plugins: [
    tailwindcss(),
  ],
}). Finally to your css files that use tailwind like the index.css and App.css if you are using them, add this line (@import "tailwindcss";) without the brackets. Thats it and rerun the server. then you can use your tailwind in your html or pages that you write.

**Reply by rozsazoltan** [reply]
It was too old question for v3.

**Comment by Kvngstar** [comment]
@na-zi-ya did you import

```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

In your index.css file? If not do so

**Reply by na-zi-ya** [reply]
> @na-zi-ya did you import
> 
> ```
> @tailwind base;
> @tailwind components;
> @tailwind utilities;
> ```
> 
> In your index.css file? If not do so

Yes..

**Reply by myselfsiddharth** [reply]
I am in the same boat. @na-zi-ya were you able to fix it ?
I am not quite sure what is going wrong

**Reply by ChauManh** [reply]
https://github.com/tailwindlabs/tailwindcss/discussions/12312#discussioncomment-13186323

**Comment by gptkong** [comment]
![image](https://github.com/user-attachments/assets/a765e8f0-26f7-4262-84ae-874e504903d2)

My tailwindcss is also not working, may I ask why?

Framework: vite+react-ts

here is package.json
```json
{
  "name": "vite-react-ts-map",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc -b && vite build",
    "lint": "eslint .",
    "preview": "vite preview"
  },
  "dependencies": {
    "mapbox-gl": "^3.7.0",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@eslint/js": "^9.9.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "@vitejs/plugin-react": "^4.3.1",
    "autoprefixer": "^10.4.20",
    "eslint": "^9.9.0",
    "eslint-plugin-react-hooks": "^5.1.0-rc.0",
    "eslint-plugin-react-refresh": "^0.4.9",
    "globals": "^15.9.0",
    "postcss": "^8.4.47",
    "tailwindcss": "^3.4.13",
    "typescript": "^5.5.3",
    "typescript-eslint": "^8.0.1",
    "vite": "^5.4.1"
  }
}

```

**Comment by Kvngstar** [comment]
In the vite.config.js, copy and paste this

```
 import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import tailwindcss from 'tailwindcss'


export default defineConfig({
  plugins: [react()],
  css: {
    postcss: {
      plugins: [tailwindcss()],
    },
  }
})
```
This is the solution provided earlier, try this @gptkong  @na-zi-ya

**Reply by HishamAbualhaj** [reply]
Helped me so much, Thank a lot

**Reply by Kvngstar** [reply]
You're welcome @HishamAbualhaj

**Reply by W1Real** [reply]
Worked here as well.

Debug info for the devs (this is how I installed):
```
pnpm create vite
# 1) Type name
# 2) Select "TypeScript + SWC"
# 3) Done.

pnpm install -D tailwindcss@latest postcss@latest autoprefixer@latest
pnpm dlx tailwindcss init

pnpm install tailwind-merge
```

Then... imported and added the plugin like he instructed above.
OBS: Since I'm using SWC the only difference is on the import statement above, I import react from '@vitejs/plugin-react-swc'

**Reply by AmulApplore** [reply]
thanks man it just works

**Reply by Hassan3322751** [reply]
Very thanks its worked perfectly

**Comment by AbidKhan01ak** [comment]
Apparently, it worked for me by removing a forward slash from tailwind.config.js. 

`export default {
  content: [
    "./index.html",
    "./src/**/*/.{js,ts,jsx,tsx}",
  ],`

It came like this, then I removed one forward slash from the line `"./src/**/*/.{js,ts,jsx,tsx}",` and it worked.
The final code is 

`export default {
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}",
  ],`

**Comment by befabry** [comment]
Hello,

I am using **Preact**, Tailwind and Docker.  It's not working and I don't get why.

My docker-compose file
```yml
# docker-compose.yml
services:
  frontend:
    image: node:23-alpine
    restart: unless-stopped
    volumes:
      - '.:/app'
    ports:
      - '5173:5173'
    environment:
      - NODE_ENV=development
    command: ['npm', 'run', 'dev']
    working_dir: /app
    networks:
      - caddie_network

networks:
  caddie_network:
    external: true
      #name: caddie_network
```
my postcss config
```js
// postcss.config.mjs
export default {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
};

```

my tailwind config
```js
// tailwind.config.mjs
/** @type {import('tailwindcss').Config} */
export default {
  content: ['./index.html', './src/**/*.{js,ts,jsx,tsx}'],
  presets: [],
  darkMode: 'media', // or 'class'
  theme: {
    extend: {
      colors: {
        text: '#cad3f5',
        subtext1: '#b8c0e0',
        subtext0: '#a5adcb',
        overlay2: '#939ab7',
        overlay1: '#8087a2',
        overlay0: '#6e738d',
        surface2: '#5b6078',
        surface1: '#494d64',
        surface0: '#363a4f',
        base: '#24273a',
        mantle: '#1e2030',
        crust: '#181926',
        yellow: '#eed49f',
        green: '#a6da95',
        sapphire: '#7dc4e4',
        teal: '#8bd5ca',
        sky: '#91d7e3',
        blue: '#8aadf4',
        lavender: '#b7bdf8',
        peach: '#f5a97f',
        maroon: '#ee99a0',
        red: '#ed8796',
        mauve: '#c6a0f6',
        pink: '#f5bde6',
        flamingo: '#f0c6c6',
        rose: '#f4dbd6',
      },
    },
  },
  plugins: [],
};
```

my vite config

```ts
//vite.config.ts

// eslint-disable-next-line import/no-named-as-default
import preact from '@preact/preset-vite';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [preact(), tsconfigPaths()],
  server: {
    proxy: {
      '/api': {
        target: 'http://backend:3001',
        changeOrigin: true,
        rewrite: (path) => path.replace(/^\/api/, ''),
        configure: (proxy) => {
          proxy.on('error', (err) => {
            console.log('proxy error', err);
          });
          proxy.on('proxyReq', (_proxyReq, req) => {
            console.log('Sending Request to the Target:', req.method, req.url);
          });
          proxy.on('proxyRes', (proxyRes, req) => {
            console.log('Received Response from the Target:', proxyRes.statusCode, req.url);
          });
        },
      },
    },
  },
});
```
I'm having to import the CDN in my index.html and some classes aren't working. It worked for about 1 evening, then it didn't.

I tried various tinkering I found here, none of them worked.

If you think it's easier to check out the repo you can have a look here :

[repo](https://github.com/La-Fabryck/caddie-frontend-preact/tree/debug)

**Reply by AbidKhan01ak** [reply]
Run this command locally to see if Tailwind is generating the styles:

`npx tailwindcss build -o output.css
`

If this fails, there’s likely an issue with the Tailwind setup or your environment.


Verify the PostCSS plugin is running:

`npx postcss --config postcss.config.mjs
`

**Reply by befabry** [reply]
the first command

```bash
npx tailwindcss build -o output.css 
```

seems to work fine.

However for the second, I can't make it work. I had to install postcss-cli

I tried through Docker and without, I'm getting :

```
> docker compose run --rm frontend  ./node_modules/.bin/postcss --config postcss.config.mjs
Input Error: Did not receive any STDIN
```

It's looking like this : 

![image](https://github.com/user-attachments/assets/1900c869-4512-4304-943f-d0c3affcf0c1)

instead of something like that :

![image](https://github.com/user-attachments/assets/a1babc9e-37c3-4c52-8cde-7b2a9263a1ff)

Would it be possible that PostCSS isn't doing what it's supposed ? I'm having trouble making it work with or without Docker

**Reply by befabry** [reply]
I had an empty presets :

```js
/** @type {import('tailwindcss').Config} */
export default {
  content: ['./index.html', './src/**/*.{js,ts,jsx,tsx}'],
  // presets: [], <----------------
  darkMode: 'media', // or 'class'
  theme: {
    extend: {
      colors: {
        text: '#cad3f5',
        subtext1: '#b8c0e0',
        subtext0: '#a5adcb',
        overlay2: '#939ab7',
        overlay1: '#8087a2',
        overlay0: '#6e738d',
        surface2: '#5b6078',
        surface1: '#494d64',
        surface0: '#363a4f',
        base: '#24273a',
        mantle: '#1e2030',
        crust: '#181926',
        yellow: '#eed49f',
        green: '#a6da95',
        sapphire: '#7dc4e4',
        teal: '#8bd5ca',
        sky: '#91d7e3',
        blue: '#8aadf4',
        lavender: '#b7bdf8',
        peach: '#f5a97f',
        maroon: '#ee99a0',
        red: '#ed8796',
        mauve: '#c6a0f6',
        pink: '#f5bde6',
        flamingo: '#f0c6c6',
        rose: '#f4dbd6',
      },
    },
  },
  plugin
```

See answer here : https://github.com/tailwindlabs/tailwindcss/discussions/15373#discussioncomment-11536584

**Comment by TomerH0525** [comment]
no solutions here worked for me ...
ive got this setup 
![image](https://github.com/user-attachments/assets/c1cb48f9-8734-43a1-b5ff-ee3d467a21a1)
running npm run dev will throw the following error :
![image](https://github.com/user-attachments/assets/63d51d2c-aafb-4db5-9d9b-ee4a1206b602)

**Reply by TomerH0525** [reply]
figured it out...
in the tailwind.config.js instead of module.exports = {......
i typed export default {....
and it worked...

**Comment by fcFn** [comment]
Since I'm using an Nx monorepo, I had to point vite for my app to the correct config file when initializing the tailwindcss plugin: 

```ts
// apps/my-app/vite.config.ts
export default defineConfig({
...
css: {
    postcss: {
      plugins: [tailwindcss(path.resolve(__dirname, './tailwind.config.ts'))],
    },
  },
...
```

You also might need to point the `content` in the tailwind config file to your specific app directory.

All that is assuming you want to include tailwind only for a specific app in your monorepo.

**Comment by elma-webdev** [comment]
hello guys, I was looking at your discussion becouse I faced the same issue, so in my case it was missing the script " import './index.css' " in my main.tsx! @rmGFW

**Comment by atcherdsd** [comment]
The documentation provides a new import method:

@import "tailwindcss";

**Reply by rmelara-designli** [reply]
what does that mean? where should I put that @import?

**Reply by rmelara-designli** [reply]
I got it , this config work for me!

here is my config using vite and  "@tailwindcss/vite": "^4.1.3",

vite.config.ts
```ts
import tailwindcss from '@tailwindcss/vite'
import { defineConfig } from 'vite'

export default defineConfig({
  plugins: [
    tailwindcss(),
  ],
})
```


index.css
```css
@import "tailwindcss";
@tailwind base;
@tailwind components;
@tailwind utilities;
```


main.tsx
```tsx
import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import './index.css'

import App from './App.tsx'

createRoot(document.getElementById('root')!).render(
  <StrictMode>
    <App />
  </StrictMode>,
)
```


package json dependencies and dev dependencies
```json
"dependencies": {
  "@heroui/react": "^2.7.6",
  "@tailwindcss/vite": "^4.1.3",
  "framer-motion": "^12.6.5",
  "lucide-react": "^0.487.0",
  "react": "^19.0.0",
  "react-dom": "^19.0.0"
},
"devDependencies": {
  "@eslint/js": "^9.21.0",
  "@types/react": "^19.0.10",
  "@types/react-dom": "^19.0.4",
  "@vitejs/plugin-react": "^4.3.4",
  "autoprefixer": "^10.4.21",
  "eslint": "^9.21.0",
  "eslint-plugin-react-hooks": "^5.1.0",
  "eslint-plugin-react-refresh": "^0.4.19",
  "globals": "^15.15.0",
  "typescript": "~5.7.2",
  "typescript-eslint": "^8.24.1",
  "vite": "^6.2.0"
}
```

**Comment by MeerMuneeb** [comment]
# ⚡️ React + Vite + Tailwind CSS Boilerplate

An error-free, ready-to-use starter template for React with Vite and Tailwind CSS.  
Tailwind setup has been tested and fixed to avoid common configuration issues.  
Clone this boilerplate and start building right away — no more setup headaches 🚀.

---

## Features

✅ Vite — Lightning-fast build & HMR  
✅ React 18  
✅ Tailwind CSS 3  
✅ PostCSS  
✅ Autoprefixer  
✅ Clean project structure  
✅ Zero errors — fully tested starter  
✅ Minimal dependencies

---

## Quick Start

```bash
# Clone the repo
git clone https://github.com/your-username/vite-react-tailwind-boilerplate.git

# Navigate to project folder
cd vite-react-tailwind-boilerplate

# Install dependencies
npm install

# Start development server
npm run dev

**Reply by rozsazoltan** [reply]
Sorry, but I don't understand why I should clone a repo. My entire Vite + React + Tailwind CSS v4 starter setup is ready with just two commands.

Follow Vite create-app steps: https://vite.dev/guide/#scaffolding-your-first-vite-project

```none
npm create vite@latest my-app-name

? Select a framework: › - Use arrow-keys. Return to submit.
    Vanilla
    Vue
❯   React
    Preact
    Lit
    Svelte
   Others

# Enter the project to download dependencies
cd my-electron-vite-project
npm install

# Install Tailwind CSS v4
npm install tailwindcss @tailwindcss/vite
```

Follow Tailwind CSS v4 installation steps: https://tailwindcss.com/docs/installation/using-vite

**vite.config.js**
```css
import { defineConfig } from 'vite'
import tailwindcss from '@tailwindcss/vite'
export default defineConfig({
  plugins: [
    tailwindcss(),
  ],
})
```

**global.css**
```css
@import "tailwindcss";
```

Now run project:
```none
npm run dev
```

**Comment by watashigoto** [comment]
The question was for version 3. 

I have tips for those who are looking for **version 4** which is the latest version currently.

Just follow the guide here: https://tailwindcss.com/docs/installation/using-vite

It's really simple and no need to create or edit any postcss config or tailwind config.

**If it still doesn't work**

Open the package.json file and check the tailwindcss version. For some reason the installed @tailwindcss/vite was version 4 but tailwindcss was version 3. You need to upgrade the tailwindcss version.

Instead of using this command

`npm install tailwindcss @tailwindcss/vite`

use this one

`npm install tailwindcss@4 @tailwindcss/vite`

It should work now.

**Reply by rozsazoltan** [reply]
There's no problem with the `npm install tailwindcss @tailwindcss/vite` command - the latest version is v4, so currently `npm install tailwindcss` is equivalent to `npm install tailwindcss@4`. It would be better to educate developers to check that the latest version is indeed the one they want to install - usually it is, but they often don't check the documentation for a while and miss the breaking changes.

So you need to learn how to manage dependencies in an existing project. The `npm update` command is perfect for this. And here, for example, if I had TailwindCSS v2, it would be good to first update version-specifically to v3 with `npm install tailwindcss@3` (since v3 not latest version, should use version-specific cmd), and then reach the latest version with `npm update tailwindcss`. Of course, it would be crazy to migrate directly from v2 to v4, but some people feel it's necessary.
* [**Node update a specific package**](https://stackoverflow.com/a/43127994/15167500)
* [How can I update each dependency in package.json to the latest version?](https://stackoverflow.com/a/16074029/15167500)

```none
npm update tailwindcss
npm install tailwindcss @tailwindcss/vite
```

Firstly for new projects:

Same:
* [Deprecated CRA; How to install Next.js; How to install Vite with React.js](https://stackoverflow.com/a/79650919/15167500)
* [Electron-Vite + React + Tailwind CSS v4](https://stackoverflow.com/a/79562636/15167500)
* [Issue with Loading HeroUI Styles in Electron with Vite, Tailwind CSS v4, and React](https://stackoverflow.com/a/79742605/15167500)
* [Install Tailwind CSS with React.js](https://rozsazoltan.vercel.app/docs/tailwindcss/installation/framework-guides/react/vite)

And to better understand v4:

Related:
* [Upgrade guide](https://tailwindcss.com/docs/upgrade-guide)
* [Theme variables (configuration)](https://tailwindcss.com/docs/theme)
* [What's breaking changes from v4?](https://stackoverflow.com/a/79383884/15167500)
* [Tailwind CSS v4: more packages and new Vite support](https://rozsazoltan.vercel.app/blog/2025-05-20-tailwind-css-v4-separated-packages)
* [How to switch to a CSS-first configuration in Tailwind CSS v4 and above](https://rozsazoltan.vercel.app/blog/2025-05-25-tailwind-css-v4-css-first-configuration)

Other related:
* [How to use custom color themes in TailwindCSS v4](https://stackoverflow.com/a/79499827/15167500)
* [How to override theme variables in TailwindCSS v4 - `@theme` vs `@layer theme` vs `:root`](https://stackoverflow.com/a/79691838/15167500)
* [Should I use `@theme` or `@theme inline`?](https://stackoverflow.com/a/79705934/15167500)
* [When should I use `*` and when should I use `:root, :host` as the parent selector?](https://stackoverflow.com/a/79692653/15167500)
